Convert controller disruption and endpoint files to structured logs

scottilee opened the pull request:
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

scottilee committed:
Convert controller disruption and endpoint files to structured logs

scottilee commented:
/remove-sig instrumentation
/area logging

danwinship commented on pkg/controller/disruption/disruption.go:
```diff
@@ -381,14 +381,14 @@ func (dc *DisruptionController) Run(stopCh <-chan struct{}) {
 func (dc *DisruptionController) addDb(obj interface{}) {
 	pdb := obj.(*policy.PodDisruptionBudget)
-	klog.V(4).Infof("add DB %q", pdb.Name)
+	klog.V(4).InfoS("Add DB", "pdb", pdb.Name)
```
api objects are supposed to be logged with `klog.KObj()`, right?

```suggestion
	klog.V(4).InfoS("Add DB", "pdb", klog.KObj(pdb))
```

(likewise throughout)

scottilee replied:
Correct, fixed!

danwinship commented on pkg/controller/disruption/disruption.go:
```diff
@@ -397,40 +397,40 @@ func (dc *DisruptionController) removeDb(obj interface{}) {
 		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 		if !ok {
-			klog.Errorf("Couldn't get object from tombstone %+v", obj)
+			klog.ErrorS(nil, "Couldn't get object from tombstone", "obj", obj)
```
"With the exception of acronyms like "IP" and the standard "err", don't shorten names."
```suggestion
			klog.ErrorS(nil, "Couldn't get object from tombstone", "object", obj)
```

scottilee replied:
Fixed.

danwinship commented on pkg/controller/endpoint/endpoints_controller.go:
```diff
@@ -186,8 +186,8 @@ func (e *Controller) Run(workers int, stopCh <-chan struct{}) {
-	klog.Infof("Starting endpoint controller")
-	defer klog.Infof("Shutting down endpoint controller")
+	klog.InfoS("Starting Endpoint controller")
+	defer klog.InfoS("Shutting down Endpoint controller")
```
if you're capitalizing "Endpoint" because it's a type, then the actual type name is "Endpoints" with an "s", not "Endpoint"

scottilee replied:
Fixed.

danwinship commented on pkg/controller/endpoint/endpoints_controller.go:
```diff
@@ -441,19 +441,19 @@ func (e *Controller) syncService(key string) error {
 	for _, pod := range pods {
 		if len(pod.Status.PodIP) == 0 {
-			klog.V(5).Infof("Failed to find an IP for pod %s/%s", pod.Namespace, pod.Name)
+			klog.V(5).InfoS("Failed to find an IP for Pod", "podNamespace", pod.Namespace, "pod", pod.Name)
```
```suggestion
			klog.V(5).InfoS("Failed to find an IP for Pod", "pod", klog.KObj(pod))
```

etc

scottilee replied:
Fixed.

scottilee commented:
@danwinship Thanks for the review! I addressed your comments about:
* Not shortening "object"
* Using klog.KObj()
* Using correct object name "Endpoints"

yangjunmyfm192085 commented on pkg/controller/disruption/disruption.go:
```diff
@@ -397,40 +397,40 @@ func (dc *DisruptionController) removeDb(obj interface{}) {
 func (dc *DisruptionController) addPod(obj interface{}) {
 	pod := obj.(*v1.Pod)
-	klog.V(4).Infof("addPod called on pod %q", pod.Name)
+	klog.V(4).InfoS("addPod called on Pod", "pod", klog.KObj(pod))
```
```suggestion
	klog.V(4).InfoS("AddPod called on Pod", "pod", klog.KObj(pod))
```

scottilee replied:
Fixed.

yangjunmyfm192085 commented on pkg/controller/disruption/disruption.go:
```diff
@@ -397,40 +397,40 @@ func (dc *DisruptionController) removeDb(obj interface{}) {
 		return
 	}
-	klog.V(4).Infof("addPod %q -> PDB %q", pod.Name, pdb.Name)
+	klog.V(4).InfoS("addPod", "pod", klog.KObj(pod), "pdb", klog.KObj(pdb))
```
```suggestion
	klog.V(4).InfoS("AddPod", "pod", klog.KObj(pod), "pdb", klog.KObj(pdb))
```

scottilee replied:
Fixed.

yangjunmyfm192085 commented on pkg/controller/disruption/disruption.go:
```diff
@@ -397,40 +397,40 @@ func (dc *DisruptionController) removeDb(obj interface{}) {
 func (dc *DisruptionController) updatePod(old, cur interface{}) {
 	pod := cur.(*v1.Pod)
-	klog.V(4).Infof("updatePod called on pod %q", pod.Name)
+	klog.V(4).InfoS("updatePod called on Pod", "pod", klog.KObj(pod))
```
```suggestion
	klog.V(4).InfoS("UpdatePod called on Pod", "pod", klog.KObj(pod))
```

scottilee replied:
Fixed.

yangjunmyfm192085 commented on pkg/controller/disruption/disruption.go:
```diff
@@ -397,40 +397,40 @@ func (dc *DisruptionController) removeDb(obj interface{}) {
 		return
 	}
-	klog.V(4).Infof("updatePod %q -> PDB %q", pod.Name, pdb.Name)
+	klog.V(4).InfoS("updatePod", "pod", klog.KObj(pod), "pdb", klog.KObj(pdb))
```
```suggestion
	klog.V(4).InfoS("UpdatePod", "pod", klog.KObj(pod), "pdb", klog.KObj(pdb))
```

scottilee replied:
Fixed.

yangjunmyfm192085 commented on pkg/controller/disruption/disruption.go:
```diff
@@ -444,29 +444,29 @@ func (dc *DisruptionController) deletePod(obj interface{}) {
 		}
 	}
-	klog.V(4).Infof("deletePod called on pod %q", pod.Name)
+	klog.V(4).InfoS("deletePod called on Pod", "pod", klog.KObj(pod))
```
```suggestion
	klog.V(4).InfoS("DeletePod called on Pod", "pod", klog.KObj(pod))
```

scottilee replied:
Fixed.

yangjunmyfm192085 commented on pkg/controller/disruption/disruption.go:
```diff
@@ -444,29 +444,29 @@ func (dc *DisruptionController) deletePod(obj interface{}) {
 		return
 	}
-	klog.V(4).Infof("deletePod %q -> PDB %q", pod.Name, pdb.Name)
+	klog.V(4).InfoS("deletePod", "pod", klog.KObj(pod), "pdb", klog.KObj(pdb))
```
```suggestion
	klog.V(4).InfoS("DeletePod", "pod", klog.KObj(pod), "pdb", klog.KObj(pdb))
```

scottilee replied:
Fixed.

yangjunmyfm192085 commented on pkg/controller/disruption/disruption.go:
```diff
@@ -578,7 +578,7 @@ func (dc *DisruptionController) sync(key string) error {
 	}
 	if err != nil {
-		klog.Errorf("Failed to sync pdb %s/%s: %v", pdb.Namespace, pdb.Name, err)
+		klog.ErrorS(err, "Failed to sync PodDisruptionBudget", "pdbNamespace", pdb.Namespace, "pdb", pdb.Name)
```
klog.KObj or klog.KRef?

scottilee replied:
Fixed.

yangjunmyfm192085 commented on pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go:
```diff
@@ -399,7 +399,7 @@ func (c *Controller) onEndpointsAdd(obj interface{}) {
 	}
 	if !c.shouldMirror(endpoints) {
-		klog.V(5).Infof("Skipping mirroring for %s/%s", endpoints.Namespace, endpoints.Name)
+		klog.V(5).Infof("Skipping mirroring", "endpoints", klog.KObj(endpoints))
```
```suggestion
		klog.V(5).InfoS("Skipping mirroring", "endpoints", klog.KObj(endpoints))
```

scottilee replied:
Fixed.

scottilee committed:
Address comments and add to .structured_logging

Restore removed line

Fixes

scottilee commented:
@yangjunmyfm192085 Thanks for the review! Let me know if there's anything else.

yangjunmyfm192085 commented:
/ok-to-test

scottilee commented:
Restored an accidentally deleted line that caused the test failures.

scottilee commented:
Addressed the "pull-kubernetes-verify" errors.

scottilee commented:
/retest

yangjunmyfm192085 commented:
/wg structured-logging
part duplicated with https://github.com/kubernetes/kubernetes/pull/100672
